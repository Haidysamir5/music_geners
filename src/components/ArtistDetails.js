import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

class ArtistDetails extends Component {
  render() {
    const { artist } = this.props;
    const { id, name, picture_medium, type, tracklist } = artist;
    return (
      <div className="col-12 col-md-3 col-xl-2">
        <div className="single-podcast-area mb-30 wow fadeInUp">
          {/* <!-- Thumbnail --> */}
          <div className="podcast-thumb">
            <img src={picture_medium} alt="" />
          </div>
          {/* <!-- Content --> */}
          <div className="podcast-content">
            <div className="podcast-meta">{type}</div>
            <h6>{name}</h6>
          </div>
        </div>
      </div>
    );
  }
}

export default connect()(ArtistDetails);
