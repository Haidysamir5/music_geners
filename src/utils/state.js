export const music={
    modal_state:false,
    music_info:[]
}