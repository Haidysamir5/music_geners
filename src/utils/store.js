import { createStore, applyMiddleware } from "redux";
import promiseMiddleware from "redux-promise";
import reducers from "../reducers";
export const store = createStore(reducers, applyMiddleware(promiseMiddleware));

