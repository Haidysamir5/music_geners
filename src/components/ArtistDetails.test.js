import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import ArtistDetails from "./ArtistDetails";
// wrap the root component in a Provider
import { store } from "../utils/store";

test("match snapshoot", () => {
  const wrapper = shallow(<ArtistDetails store={store} />);
  
  expect(toJson(wrapper)).toMatchSnapshot();
});


