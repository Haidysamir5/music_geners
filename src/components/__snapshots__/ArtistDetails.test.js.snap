// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshoot 1`] = `
<ArtistDetails
  dispatch={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
